### PR TITLE
Dev/iliaskhan/update blob api version

### PR DESCRIPTION
**Pull request by IliasKhan**
## ARM API Information (Control Plane)

<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow.
</i>
Azure 1st Party Service can try out the [Shift Left](https://aka.ms/ShiftLeft) experience to initiate API design review from ADO code repo.  If you are interested, may request engineering support by filling in with the form https://aka.ms/ShiftLeftSupportForm.

### Changelog
Add a changelog entry for this PR by answering the following questions:
  1. What's the purpose of the update?
      - [ ] new service onboarding
      - [x] new API version
      - [ ] update existing version for new feature
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify
  2. When are you targeting to deploy the new service/feature to public regions? Please provide the date or, if the date is not yet available, the month.
  3. When do you expect to publish the swagger? Please provide date or, the the date is not yet available, the month.
  4. By default, Azure SDKs of all languages (.NET/Python/Java/JavaScript for both management-plane SDK and data-plane SDK, Go for management-plane SDK only ) MUST be refreshed with/after swagger of new version is published. If you prefer NOT to refresh any specific SDK language upon swagger updates in the current PR, please leave details with justification here.

### Contribution checklist (MS Employees Only):
- [ ] I commit to follow the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy) of "no breaking changes"
- [ ] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [ ] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist

> **Applicability**: :warning:
>
> If your changes encompass only the following scenarios, you should SKIP this section, as these scenarios do not require ARM review.
> - Change to data plane APIs
> - Adding new properties
> - All removals

Otherwise your PR may be subject to ARM review requirements. Complete the following:
- [ ] Check this box if any of the following apply to the PR so that the label "ARMReview" and "WaitForARMFeedback" will be added by bot to kick off ARM API Review.  Missing to check this box in the following scenario may result in delays to the ARM manifest review and deployment.
  - Adding a new service
  - Adding new API(s)
  - Adding a new API version
    -[ ] To review changes efficiently, ensure you copy the existing version into the new directory structure for first commit and then push new changes, including version updates, in separate commits. You can use OpenAPIHub to initialize the PR for adding a new version. For more details refer to the [wiki](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/208/OpenAPI-Hub-Adding-new-API-version).

- [ ] Ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them.

### Breaking Change Review Checklist
If you have any breaking changes as defined in the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy/), request approval from the Breaking Change Review Board.

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Additional details on the process and office hours are on the [Breaking Change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

NOTE: To update API(s) in public preview for over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Comment by tadelesh** [comment]
@iliaskha I noticed that `./Microsoft.Sql/preview/2020-11-01-preview/ServerDevOpsAudit.json` has been removed. Could you help to explain the reason?

File: specification/sql/resource-manager/readme.md [hunk 1]
```diff
@@ -563,14 +563,7 @@ APIs must only be added to this section when the API is publicly available in at
 
 ``` yaml $(tag) == 'package-composite-v5'
 input-file:
-- Microsoft.Sql/stable/2014-04-01-legacy/backups_legacy.json
```
Comment (ericshape) [inline specification/sql/resource-manager/readme.md:566]: why do we move these files to the bottom?

File: specification/sql/resource-manager/readme.md [hunk 2]
```diff
@@ -579,9 +572,8 @@ input-file:
 - ./Microsoft.Sql/preview/2020-11-01-preview/DatabaseSecurityAlertPolicies.json
 - ./Microsoft.Sql/preview/2020-11-01-preview/DatabaseTables.json
 - ./Microsoft.Sql/preview/2020-11-01-preview/DatabaseVulnerabilityAssesmentRuleBaselines.json
-- ./Microsoft.Sql/preview/2020-11-01-preview/DatabaseVulnerabilityAssessments.json
 - ./Microsoft.Sql/preview/2020-11-01-preview/DatabaseVulnerabilityAssessmentScans.json
-- ./Microsoft.Sql/preview/2020-11-01-preview/DataWarehouseUserActivities.json
```
Comment (ericshape) [inline specification/sql/resource-manager/readme.md:584]: why do we delete this user activity json?

File: specification/sql/resource-manager/readme.md [hunk 3]
```diff
@@ -630,11 +623,12 @@ input-file:
 - ./Microsoft.Sql/preview/2020-11-01-preview/PrivateLinkResources.json
 - ./Microsoft.Sql/preview/2020-11-01-preview/RecoverableManagedDatabases.json
 - ./Microsoft.Sql/preview/2020-11-01-preview/RestorePoints.json
+- ./Microsoft.Sql/preview/2020-11-01-preview/SensitivityLabels.json
 - ./Microsoft.Sql/preview/2020-11-01-preview/ServerAdvisors.json
 - ./Microsoft.Sql/preview/2020-11-01-preview/ServerAutomaticTuning.json
 - ./Microsoft.Sql/preview/2020-11-01-preview/ServerAzureADAdministrators.json
 - ./Microsoft.Sql/preview/2020-11-01-preview/ServerAzureADOnlyAuthentications.json
-- ./Microsoft.Sql/preview/2020-11-01-preview/ServerDevOpsAudit.json
+- ./Microsoft.Sql/preview/2022-02-01-preview/ServerDevOpsAudit.json
```
Comment (ericshape) [inline specification/sql/resource-manager/readme.md:631]: can we sort the json file paths in alphabetical order?

**Comment by IliasKhan** [comment]
> @IliasKhan I noticed that `./Microsoft.Sql/preview/2020-11-01-preview/ServerDevOpsAudit.json` has been removed. Could you help to explain the reason?

I just updated the api version to 2022-02-01

**Review by ericshape — APPROVED** [review]
LGTM

**Comment by tadelesh** [comment]
Go berakings are from previous PR and lack of `-alwaysSetBodyParamRequired` in `build.go`. Approved and cc @Alancere .

**Comment by xiaoxuqi-ms** [comment]
This PR's required for User Managed Identity release, Management Service changes already deployed, Portal changes are going to deploy.

there will be incosistency if we don't deploy Powershell and CLI changes.

**Comment by tadelesh** [comment]
Regarding Go breaking changes, I listed all related swagger change and breaking reasons as follows. @xiaoxuqi-ms, I haven't found related breaking changes review info, and could you help to check?
All swagger file changes:
```
Old
- ./Microsoft.Sql/preview/2020-11-01-preview/ManagedDatabaseRestoreDetails.json
- ./Microsoft.Sql/preview/2020-11-01-preview/ManagedDatabases.json
- ./Microsoft.Sql/preview/2020-11-01-preview/ServerDevOpsAudit.json
- ./Microsoft.Sql/preview/2020-11-01-preview/VirtualClusters.json
- ./Microsoft.Sql/preview/2021-05-01-preview/Databases.json
- ./Microsoft.Sql/preview/2021-11-01-preview/ReplicationLinks.json

New
- ./Microsoft.Sql/preview/2022-02-01-preview/ManagedDatabaseRestoreDetails.json
- ./Microsoft.Sql/preview/2022-02-01-preview/ManagedDatabases.json
- ./Microsoft.Sql/preview/2022-02-01-preview/ServerDevOpsAudit.json
- ./Microsoft.Sql/preview/2022-05-01-preview/VirtualClusters.json
- ./Microsoft.Sql/preview/2022-02-01-preview/Databases.json
- ./Microsoft.Sql/preview/2022-02-01-preview/ReplicationLinks.json
```
Go breaking reasons:
```
Non-LRO -> LRO
ReplicationLinks.json
- Function `*ReplicationLinksClient.Delete` has been removed
- Struct `ReplicationLinksClientDeleteOptions` has been removed
VirtualClusters.json
- Function `*VirtualClustersClient.UpdateDNSServers` has been removed
- Struct `VirtualClustersClientUpdateDNSServersOptions` has been removed

Property removal
VirtualClusters.json
- Field `Family` of struct `VirtualClusterProperties` has been removed
- Field `MaintenanceConfigurationID` of struct `VirtualClusterProperties` has been removed

Add x-ms-enum
ServerDevOpsAudit.json
- Function `*ServerDevOpsAuditSettingsClient.Get` parameter(s) have been changed from `(context.Context, string, string, string, *ServerDevOpsAuditSettingsClientGetOptions)` to `(context.Context, string, string, DevOpsAuditingSettingsName, *ServerDevOpsAuditSettingsClientGetOptions)`
- Function `*ServerDevOpsAuditSettingsClient.BeginCreateOrUpdate` parameter(s) have been changed from `(context.Context, string, string, string, ServerDevOpsAuditingSettings, *ServerDevOpsAuditSettingsClientBeginCreateOrUpdateOptions)` to `(context.Context, string, string, DevOpsAuditingSettingsName, ServerDevOpsAuditingSettings, *ServerDevOpsAuditSettingsClientBeginCreateOrUpdateOptions)`

Property type change
ManagedDatabaseRestoreDetails.json
- Type of `ManagedDatabaseRestoreDetailsProperties.UnrestorableFiles` has been changed from `[]*string` to `[]*ManagedDatabaseRestoreDetailsUnrestorableFileProperties`
- Type of `ManagedDatabaseRestoreDetailsProperties.NumberOfFilesDetected` has been changed from `*int64` to `*int32`
- Type of `ManagedDatabaseRestoreDetailsProperties.PercentCompleted` has been changed from `*float64` to `*int32`

Response model change
VirtualClusters.json
- Field `UpdateManagedInstanceDNSServersOperation` of struct `VirtualClustersClientUpdateDNSServersResponse` has been removed
- Struct `UpdateManagedInstanceDNSServersOperation` has been removed
- Struct `DNSRefreshConfigurationProperties` has been removed
- Const `DNSRefreshConfigurationPropertiesStatusSucceeded` has been removed
- Const `DNSRefreshConfigurationPropertiesStatusFailed` has been removed
- Type alias `DNSRefreshConfigurationPropertiesStatus` has been removed
- Function `PossibleDNSRefreshConfigurationPropertiesStatusValues` has been removed
```

**Comment by ericshape** [comment]
> Regarding Go breaking changes, I listed all related swagger change and breaking reasons as follows. @xiaoxuqi-ms, for this PR, I think the adding of `x-ms-enum` should be reviewed. All swagger file changes:
> 
> ```
> Old
> - ./Microsoft.Sql/preview/2020-11-01-preview/ManagedDatabaseRestoreDetails.json
> - ./Microsoft.Sql/preview/2020-11-01-preview/ManagedDatabases.json
> - ./Microsoft.Sql/preview/2020-11-01-preview/ServerDevOpsAudit.json
> - ./Microsoft.Sql/preview/2020-11-01-preview/VirtualClusters.json
> - ./Microsoft.Sql/preview/2021-05-01-preview/Databases.json
> - ./Microsoft.Sql/preview/2021-11-01-preview/ReplicationLinks.json
> 
> New
> - ./Microsoft.Sql/preview/2022-02-01-preview/ManagedDatabaseRestoreDetails.json
> - ./Microsoft.Sql/preview/2022-02-01-preview/ManagedDatabases.json
> - ./Microsoft.Sql/preview/2022-02-01-preview/ServerDevOpsAudit.json
> - ./Microsoft.Sql/preview/2022-05-01-preview/VirtualClusters.json
> - ./Microsoft.Sql/preview/2022-02-01-preview/Databases.json
> - ./Microsoft.Sql/preview/2022-02-01-preview/ReplicationLinks.json
> ```
> 
> Go breaking reasons:
> 
> ```
> Non-LRO -> LRO
> ReplicationLinks.json
> - Function `*ReplicationLinksClient.Delete` has been removed
> - Struct `ReplicationLinksClientDeleteOptions` has been removed
> VirtualClusters.json
> - Function `*VirtualClustersClient.UpdateDNSServers` has been removed
> - Struct `VirtualClustersClientUpdateDNSServersOptions` has been removed
> 
> Property removal
> VirtualClusters.json
> - Field `Family` of struct `VirtualClusterProperties` has been removed
> - Field `MaintenanceConfigurationID` of struct `VirtualClusterProperties` has been removed
> 
> Add x-ms-enum
> ServerDevOpsAudit.json
> - Function `*ServerDevOpsAuditSettingsClient.Get` parameter(s) have been changed from `(context.Context, string, string, string, *ServerDevOpsAuditSettingsClientGetOptions)` to `(context.Context, string, string, DevOpsAuditingSettingsName, *ServerDevOpsAuditSettingsClientGetOptions)`
> - Function `*ServerDevOpsAuditSettingsClient.BeginCreateOrUpdate` parameter(s) have been changed from `(context.Context, string, string, string, ServerDevOpsAuditingSettings, *ServerDevOpsAuditSettingsClientBeginCreateOrUpdateOptions)` to `(context.Context, string, string, DevOpsAuditingSettingsName, ServerDevOpsAuditingSettings, *ServerDevOpsAuditSettingsClientBeginCreateOrUpdateOptions)`
> 
> Property type change
> ManagedDatabaseRestoreDetails.json
> - Type of `ManagedDatabaseRestoreDetailsProperties.UnrestorableFiles` has been changed from `[]*string` to `[]*ManagedDatabaseRestoreDetailsUnrestorableFileProperties`
> - Type of `ManagedDatabaseRestoreDetailsProperties.NumberOfFilesDetected` has been changed from `*int64` to `*int32`
> - Type of `ManagedDatabaseRestoreDetailsProperties.PercentCompleted` has been changed from `*float64` to `*int32`
> 
> Response model change
> VirtualClusters.json
> - Field `UpdateManagedInstanceDNSServersOperation` of struct `VirtualClustersClientUpdateDNSServersResponse` has been removed
> - Struct `UpdateManagedInstanceDNSServersOperation` has been removed
> - Struct `DNSRefreshConfigurationProperties` has been removed
> - Const `DNSRefreshConfigurationPropertiesStatusSucceeded` has been removed
> - Const `DNSRefreshConfigurationPropertiesStatusFailed` has been removed
> - Type alias `DNSRefreshConfigurationPropertiesStatus` has been removed
> - Function `PossibleDNSRefreshConfigurationPropertiesStatusValues` has been removed
> ```

via git blame in the README.md file, I retrieved the PR for these breaking changes:

> New
> - ./Microsoft.Sql/preview/2022-02-01-preview/ManagedDatabaseRestoreDetails.json
https://github.com/Azure/azure-rest-api-specs/commit/fa8aaba152b1bb5b6fb1b2d845acb5569a5a6b56

> - ./Microsoft.Sql/preview/2022-02-01-preview/ManagedDatabases.json
https://github.com/Azure/azure-rest-api-specs/commit/fa8aaba152b1bb5b6fb1b2d845acb5569a5a6b56

> - ./Microsoft.Sql/preview/2022-02-01-preview/ServerDevOpsAudit.json
@IliasKhan is the author to remove this API

> - ./Microsoft.Sql/preview/2022-05-01-preview/VirtualClusters.json
https://github.com/Azure/azure-rest-api-specs/pull/20469

> - ./Microsoft.Sql/preview/2022-02-01-preview/Databases.json
https://github.com/Azure/azure-rest-api-specs/commit/b55607c879f1f995be07f6adcf7616772b9eca62

> - ./Microsoft.Sql/preview/2022-02-01-preview/ReplicationLinks.json
https://github.com/Azure/azure-rest-api-specs/commit/b55607c879f1f995be07f6adcf7616772b9eca62

**Comment by tadelesh** [comment]
Breaking changes are from two PRs: https://github.com/Azure/azure-rest-api-specs/pull/19820 and https://github.com/Azure/azure-rest-api-specs/pull/20469. The later one has been approved. But the former one seems not. The following breakings from this PR need to be reviewed:
```
Non-LRO -> LRO
ReplicationLinks.json
- Function `*ReplicationLinksClient.Delete` has been removed
- Struct `ReplicationLinksClientDeleteOptions` has been removed

Add x-ms-enum
ServerDevOpsAudit.json
- Function `*ServerDevOpsAuditSettingsClient.Get` parameter(s) have been changed from `(context.Context, string, string, string, *ServerDevOpsAuditSettingsClientGetOptions)` to `(context.Context, string, string, DevOpsAuditingSettingsName, *ServerDevOpsAuditSettingsClientGetOptions)`
- Function `*ServerDevOpsAuditSettingsClient.BeginCreateOrUpdate` parameter(s) have been changed from `(context.Context, string, string, string, ServerDevOpsAuditingSettings, *ServerDevOpsAuditSettingsClientBeginCreateOrUpdateOptions)` to `(context.Context, string, string, DevOpsAuditingSettingsName, ServerDevOpsAuditingSettings, *ServerDevOpsAuditSettingsClientBeginCreateOrUpdateOptions)`

Property type change
ManagedDatabaseRestoreDetails.json
- Type of `ManagedDatabaseRestoreDetailsProperties.UnrestorableFiles` has been changed from `[]*string` to `[]*ManagedDatabaseRestoreDetailsUnrestorableFileProperties`
- Type of `ManagedDatabaseRestoreDetailsProperties.NumberOfFilesDetected` has been changed from `*int64` to `*int32`
- Type of `ManagedDatabaseRestoreDetailsProperties.PercentCompleted` has been changed from `*float64` to `*int32`

```

**Comment by ericshape** [comment]
> #19820

It is the PR I released the new API version 'cause the dev branch (https://github.com/Azure/azure-rest-api-specs/commits/dev-sql-Microsoft.Sql-2022-02-01-preview/specification) has so many conflicts we could not address. 
This PR has swiped some old commits. Thus, you have to track one more commit back.

**Comment by ericshape** [comment]
> Property type change
> ManagedDatabaseRestoreDetails.json
> - Type of `ManagedDatabaseRestoreDetailsProperties.UnrestorableFiles` has been changed from `[]*string` to `[]*ManagedDatabaseRestoreDetailsUnrestorableFileProperties`
> - Type of `ManagedDatabaseRestoreDetailsProperties.NumberOfFilesDetected` has been changed from `*int64` to `*int32`
> - Type of `ManagedDatabaseRestoreDetailsProperties.PercentCompleted` has been changed from `*float64` to `*int32`

@tadelesh  it seems you have approved the SDK breaking change in https://github.com/Azure/azure-rest-api-specs/pull/19955#issuecomment-1229789460

**Comment by ericshape** [comment]
> Non-LRO -> LRO
> ReplicationLinks.json
> - Function `*ReplicationLinksClient.Delete` has been removed
> - Struct `ReplicationLinksClientDeleteOptions` has been removed

it seems the only PR for the SDK breaking change is https://github.com/Azure/azure-rest-api-specs/pull/18894/files

**Comment by mikekistler** [comment]
The tools are not flagging any breaking changes in the REST API interface. I'm simply trusting the tools here as it would be difficult to determine the effect of the change in the readme.md.

**Comment by tadelesh** [comment]
> The tools are not flagging any breaking changes in the REST API interface. I'm simply trusting the tools here as it would be difficult to determine the effect of the change in the readme.md.

@mikekistler For SDK breaking changes, our tool will report the breaking changes only when author changing the swagger version in default tag. Sometimes such change is separated with the actual swagger file change, just like this PR. So, I need to find the previous swagger file change PR and see whether the breaking changes have been approved.

@ericshape Thanks for the relay, I confirm with swagger team that the breaking changes of preview version in one year are not required to have been reviewed and approved. This [PR](https://github.com/Azure/azure-rest-api-specs/pull/19820) detected the breakings but report them only with warning. So, I approved the Go SDK breakings and for the new SDK release request with tag `package-composite-v5`, we will release a beta version.

**Review by xiaoxuqi-ms — APPROVED** [review]

**Pull request merged**